contracts/Company.sol

```diff
@@ -10,43 +12,44 @@ contract Company {
         ClassA
     }
 
-    address payable owner; //company admin
-    address employeeFund;
-    uint public currentSalary;
-    uint pension = 0;
-    salary_class class;
-    uint timeSinceClassChange;
-
-    //we will give address and initalSalary at time of deploying smart contract
-    //we are deploying contract when the employee is hired
-    constructor(
-        address _employeeFund,
-        uint _initSal,
-        salary_class _class
-    ) payable {
-        owner = payable(msg.sender);
-        employeeFund = _employeeFund;
-        currentSalary = _initSal;
-        class = _class;
-        timeSinceClassChange = block.timestamp;
+
+    struct Employee{
+        address account;
+        uint currentSalary;
+        salary_class class;
+        uint timeSinceClassChange;
+        uint experience;
     }
 
+    mapping(address=>Employee) public employeeList;
+    
+
+    constructor(address _fundContractAddress){
+        empFundContract = _fundContractAddress;
+    } 
+
+    //make a modifier if needed to set initial checks for repeated codes in various functions
+
     //function to increase salary
     //only deployer should be able to do this
     //set currentSalary to calculated salary
-    function increaseSalary(uint yrsOfWork, salary_class _class) external {
+    function increaseSalary(address empAccount,Employee memory _empl) external {
         require(msg.sender == owner, "Only deployer can change salary");
+        require(employeeList[empAccount] > 0,"This employee does not exist");
```

Comment:
Maybe we can use a flag field in the struct Employee, to distinguish between employees that exist or not. By default, the value of the flag field would be 0, but while adding an employee, we will set the flag.

Reply:
Done so in commit ac689c8